gdk/gdkglcontext.c

```diff
@@ -275,13 +341,14 @@ gdk_gl_context_download_texture (GdkGLContext    *context,
       if (priv->use_es)
         {
           for (i = y; i < height; i++)
-            glReadPixels (x, i, width, 1, GL_BGRA, GL_UNSIGNED_INT_8_8_8_8_REV,
-                          (unsigned char *) data + (i * stride));
+            glReadPixels (x, i, width, 1, GL_RGBA, GL_UNSIGNED_BYTE,
+                         (unsigned char *) data + (i * stride));
+            image_surface_rgba_to_argb (image_surface);
```

Comment:
@xjuan You should compile with GCC6, then you'd get the "misleading whitespace" warning. ;-)

@magcius For GL, yes: `GL_BGRA` and `GL_UNSIGNED_INT_8_8_8_8_REV` gives us ARGB; for GLES, those formats are not available, which needs the CPU-side channel flipping.